Include/internal/pycore_global_strings.h

```diff
@@ -21,7 +21,7 @@ extern "C" {
 #define STRUCT_FOR_STR(NAME, LITERAL) \
     STRUCT_FOR_ASCII_STR(LITERAL) _ ## NAME;
 #define STRUCT_FOR_ID(NAME) \
-    STRUCT_FOR_ASCII_STR(#NAME) _ ## NAME;
+    STRUCT_FOR_ASCII_STR(#NAME) __id__ ## NAME;
```

Comment:
This is necessary to avoid conflicts with Windows headers functions and field names. See https://github.com/kumaraditya303/cpython/actions/runs/3387388754/jobs/5628092850 as example conflicting `_tzname`  with https://learn.microsoft.com/en-us/cpp/c-runtime-library/daylight-dstbias-timezone-and-tzname?view=msvc-170. I have fixed the comment here and clarified a bit.

Reply:
Ah, thanks for clarifying.  From what I can tell, though, the problem is with macros rather than globals.

It looks like MSVC [*used to* have a global variable named `_tzname`](https://learn.microsoft.com/en-us/cpp/c-runtime-library/daylight-dstbias-timezone-and-tzname).  They later added a function called `_get_tzname()` (as well as a similar `__tzname()`) to do the same thing, deprecated `_tzname`, and then added something like `#define _tzname __tzname()`.  Thus, when we have `_tzname` as a field name, it gets expanded to `__tzname()` and we end up with "error C2032: '__tzname': function cannot be member of struct".

So it might be simpler to change the one outlier, using something like `_Py_DECLARE_STR(id_tzname, ")` and `_Py_STR(id_tzname)` (and a comment about why), instead of sticking with `_Py_ID()` and adding that `__id__` prefix.

---

Then again, the longer prefix does sort of future-proof other potential macro issues.  What you've done is probably the better way forward, so let's stick with that.  I recommend three changes, though:

* [ ] make the prefix even less likely to collide with macros in the future: `_py_`, `_py_id_`, or even `_py_str_` (my preference would be `_py_`)
* [ ] likewise adjust the prefix for `_Py_STR()`: `_py_`, or `_py_str_`, just to play it safe and stay consistent with `_Py_ID()`
* [ ] update the comment to `// to help avoid collisions with keywords, macros, etc.

FWIW, I think it would be best to have the same prefix for `_Py_global_strings.literals` and `_Py_global_strings.identifiers`, which would eliminate `_py_id_` as an option.  `_py_str_` is a bit redundant (and/or ambiguous) so I'd go with `_py_`.

Reply:
Okay let's go with `_py_` our tradition with underscore py :)

Reply:
Done in https://github.com/python/cpython/pull/99067/commits/729784a694fef3624709f3b87afc2b531123509b

Reply:
Thanks for the explanation.